### PR TITLE
feat: :sparkles: add `edit_package_properties()`

### LINKED ISSUE #715: Create and test `edit_package_properties()`
Signature: `edit_package_properties(path: Path, properties: json/dict) -> json/dict:`

Context: See diagram in the [Core Python functions](https://sprout.seedcase-project.org/docs/design/implementation/python-functions) post. 

Depends on:

- [x] #711 
- [x] #713 
- [x] #712

### PR DESCRIPTION
**Pull request by martonvago**
## Description

This adds the `edit_package_properties()` function.

Closes #715 

This PR needs an in-depth review.

## Checklist

- [x] Added or updated tests
- [x] Tests passed locally
- [x] Linted and formatted code
- [x] Build passed locally
- [x] Updated documentation

### CODE REVIEW

File: docs/design/implementation/python-functions.qmd [hunk 1]
```diff
@@ -103,19 +103,7 @@ fig-alt="A Plant UML schematic of the detailed code flow within the `create_pack
 ::: {.callout-note collapse="true"}
 ### `edit_package_properties(path, properties)`
 
-Because there already exists an empty `datapackage.json` file, we use
```
Comment (lwjohnst86) [inline docs/design/implementation/python-functions.qmd:106]: I moved this text over into the function docstring.

**Review by lwjohnst86 — DISMISSED** [review]
Very nicely done!! :stars: I only moved the docs over into the docstring, otherwise, amazing!

File: tests/core/test_edit_package_properties.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+def test_throws_error_if_properties_file_cannot_be_read(tmp_path, properties):
+    """Should throw JSONDecodeError if the properties file cannot be read as JSON."""
+    file_path = tmp_path / "datapackage.json"
+    file_path.write_text(",,, this is not, JSON")
+
+    with raises(JSONDecodeError):
+        edit_package_properties(file_path, properties)
+
+
+def test_throws_error_if_current_package_properties_malformed(tmp_path, properties):
```
Comment (signekb) [inline tests/core/test_edit_package_properties.py:56]:
```suggestion
def test_throws_error_if_current_package_properties_are_malformed(tmp_path, properties):
```
? 🤷‍♀️

File: tests/core/test_edit_package_properties.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    """Should throw NotPropertiesError if the current package properties are
+    malformed."""
+    package_properties = PackageProperties(name="invalid name with spaces").asdict
+    path = write_json(package_properties, tmp_path / "datapackage.json")
+
+    with raises(NotPropertiesError):
+        edit_package_properties(path, properties)
+
+
+def test_throws_error_if_new_package_properties_incorrect(properties_path):
```
Comment (signekb) [inline tests/core/test_edit_package_properties.py:66]:
```suggestion
def test_throws_error_if_new_package_properties_are_incorrect(properties_path):
```
? 🤷‍♀️

File: tests/core/test_edit_package_properties.py [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    """Should throw NotPropertiesError if the new package properties are incorrect."""
```
Comment (signekb) [inline tests/core/test_edit_package_properties.py:67]: Just noticing the difference in wording here with "incorrect" vs "malformed" on e.g., L58 above. Do you mean something different with these two words or do you use them as synonyms? If the latter, maybe use the same word? :)

**Review by signekb — CHANGES_REQUESTED** [review]
Once again: Super nice 🌟 
I have added some suggestions and comments.

File: sprout/core/edit_package_properties.py [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+from sprout.core.verify_is_file import verify_is_file
+from sprout.core.verify_package_properties import (
+    verify_package_properties,
+)
+from sprout.core.verify_properties_are_well_formed import (
+    verify_properties_are_well_formed,
+)
+
+
+def edit_package_properties(path: Path, properties: dict) -> dict:
```
[the remark this answers is not in the document]
Comment (martonvago) [reply sprout/core/edit_package_properties.py:15]: Hmm, that takes the `package_id` as argument, right? Which we don't really know in advance.
(Also, I think that function is called `path_package_properties` now, I'll change that in the docstring.)

File: tests/core/test_edit_package_properties.py (hunk 2 above)
Comment (martonvago) [reply tests/core/test_edit_package_properties.py:56]: Don't mind either way, but I do have a tendency to drop the "to be" in function names.

File: tests/core/test_edit_package_properties.py (hunk 4 above)
Comment (martonvago) [reply tests/core/test_edit_package_properties.py:67]: I'm trying to use "malformed" as the opposite of "well-formed" (so something that violates `verify_properties_are_well_formed`) and "incorrect" as something that violates `verify_package_properties`. So I guess this makes "incorrect" a broader term. Properties that are malformed or incomplete are also incorrect, but properties that are incorrect are not necessarily both malformed and incomplete. But I make no claims of having been super consistent on this.

**Review by lwjohnst86 — APPROVED** [review]

**Pull request merged**